File: src/main/java/com/avh/practicas/auth/service/AuthService.java
package com.avh.practicas.auth.service;

import com.avh.practicas.auth.dto.LoginResponse;

public interface AuthService {

    LoginResponse login(String correo, String password);

    void cambiarPassword(Long id, String nuevaPassword);

    void recuperar(String correo);

    void resetear(String token, String nuevaPassword);
}

File: frontend/src/modules/auth/services/authService.js
import http from '../../../shared/services/http';

const authService = {
  login: (correo, password) =>
    http.post('/auth/login', { correo, password }),

  recuperarPassword: (correo) =>
    http.post('/auth/recuperar', null, { params: { correo } }),

  resetearPassword: (token, nuevaPassword) =>
    http.post('/auth/resetear', null, { params: { token, nuevaPassword } }),

  cambiarPassword: (id, nuevaPassword) =>
    http.post('/auth/cambiar-password', null, { params: { id, nuevaPassword } })
};

export default authService;